src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -74,19 +86,31 @@ public AdbRunner(ILogger log, IAdbProcessManager processManager, string adbExePa
         }
     }
 
-    public void SetActiveDevice(string? deviceSerialNumber)
+    public void SetActiveDevice(AndroidDevice? device)
     {
-        _processManager.DeviceSerial = deviceSerialNumber ?? string.Empty;
-
-        _api = GetAPIVersion();
+        _processManager.DeviceSerial = device?.DeviceSerial ?? string.Empty;
+        _activeDevice = device;
 
-        _log.LogInformation($"Active Android device set to serial '{deviceSerialNumber}'");
+        if (device is null)
+        {
+            _log.LogInformation($"Active Android device unset");
+        }
+        else
+        {
+            _log.LogInformation($"Active Android device set to serial '{device.DeviceSerial}'");
+        }
     }
 
-    private int GetAPIVersion()
+    private int GetApiVersion()
     {
-        var output = RunAdbCommand(new[] { "shell", "getprop", "ro.build.version.sdk" });
-        return int.Parse(output.StandardOutput);
+        var output = RunAdbCommand(s_commandList[AdbPropertyNames.ApiVersion]);
+        var apiVersion = int.Parse(output.StandardOutput);
+        _activeDevice = (_activeDevice ?? new AndroidDevice(null!)) with

```

Comment:
Isn't `null!` here a mistake? I'm surprised that compiler allows this. It looks like an oxymoron. :)

Reply:
Yeah, this is suspicious. Probably something I didn't fix.
(I wrote the code 2 weeks ago and only found it in the branch today 😅)

Reply:
I found out I could simplify things a lot again thanks to this.